src/Aquifer.Public.API/Endpoints/Bibles/Alignments/Greek/Endpoint.cs

```diff
@@ -0,0 +1,272 @@
+        """;
+
+    public override void Configure()
+    {
+        Get("/bibles/{BibleId}/alignments/greek");
+        Options(EndpointHelpers.UnauthenticatedServerCacheInSeconds(EndpointHelpers.OneHourInSeconds));
+        Description(d => d.ProducesProblemFE());
+        Summary(s =>
+        {
+            s.Summary = "Gets a Bible's Greek alignment information.";
+            s.Description =
+                "For a given Bible and book of the Bible, returns the Bible's text along with associated Greek alignment information and Greek sense data for each verse, similar to a reverse interlinear Bible. Data is returned for all verses within the chapter and verse parameters.";
+        });
+    }
+
+    public override async Task HandleAsync(Request request, CancellationToken ct)
+    {
+        var bookCodeEnum = BibleBookCodeUtilities.IdFromCode(request.BookCode);
+        var lowerBounds = new BibleWordIdentifier(bookCodeEnum, request.StartChapter, request.StartVerse, request.StartWord).WordIdentifier;
+        var upperBounds = BibleWordIdentifier.GetUpperBoundOfWord(bookCodeEnum, request.EndChapter, request.EndVerse, request.EndWord).WordIdentifier;
+
+        var bookData = await dbContext.BibleBooks
+            .Where(bb =>
+                bb.Bible.Enabled &&
+                !bb.Bible.RestrictedLicense &&
+                bb.Bible.GreekAlignment &&
+                bb.Bible.Id == request.BibleId &&
+                bb.Code == request.BookCode.ToUpper())
+            .Select(bb => new
+            {
+                BibleId = bb.Bible.Id,
+                BibleName = bb.Bible.Name,
+                BibleAbbreviation = bb.Bible.Abbreviation,
+                BookCode = bb.Code,
+                BookName = bb.LocalizedName,
+            })
+            .FirstOrDefaultAsync(ct);
+
+        if (bookData == null)
+        {
+            await SendNotFoundAsync(ct);
+            return;
+        }
+
+        var greekAlignmentSource = await dbContext.Database.GetDbConnection()
+            .QueryFirstOrDefaultAsync<string>(new CommandDefinition(GreekAlignmentSourceQuery, new { bibleId = request.BibleId }));
+
+        if (greekAlignmentSource == null)
+        {
+            await SendNotFoundAsync(ct);
+            return;
+        }
+
+        var bibleText = (await dbContext.Database.GetDbConnection()
+                .QueryAsync<BibleTextWithGreekAlignmentForeignKeysResult>(
+                    new CommandDefinition(
+                        BibleTextWithGreekAlignmentForeignKeysQuery,
+                        new
+                        {
+                            lowerBounds,
+                            upperBounds,
+                            bibleId = request.BibleId,
+                        },
+                        cancellationToken: ct)))
+            .ToList();
+
+        List<GreekWordResult> greekWordResults = [];
+        foreach (var batch in bibleText.Select(r => r.GreekWordId).Distinct().Order().Chunk(size: 2000))
```

Comment:
The base query now only retrieves `GreekWordId`s and this second query here batches over those to fetch the actual Greek word data.  SQL has a 2100 parameter limit so I arbitrarily chose 2000 as the batch size.

This is powerful because we could cache all of this data by ID and only fetch data from the DB for future requests if those Greek Words are not yet in the cache.